src/dev/flang/be/jvm/JVM.java

```diff
@@ -1001,9 +1001,11 @@ boolean fieldExists(int field)
     var occ   = _fuir.clazzOuterClazz(field);
     var rt = _fuir.clazzResultClazz(field);
 
+    boolean isMainResultField = _fuir.clazzResultField(_fuir.mainClazzId()) == field;
+
     return _fuir.hasData(rt)       &&
       !_types.isScalar(occ)        &&
-      _types.clazzNeedsCode(field) &&
+      (_types.clazzNeedsCode(field) || isMainResultField) &&
```

Comment:
I think it makes sense to move this change to `FUIR.clazzNeedsCode()`, this should be the case for all backends!